qiskit/providers/ibmq/api_v2/websocket.py

```diff
@@ -84,10 +104,11 @@ def connect(self, url):
             auth_response = WebsocketMessage.from_bytes(auth_response_raw)
 
             if auth_response.type_ != 'authenticated':
-                raise WebsocketError(auth_response.as_json())
-        except websockets.ConnectionClosed as ex:
+                raise WebsocketIBMQProtocolError(auth_response.as_json())
```

Comment:
Mostly because of the protocol design - if we send an invalid token, the IBM Q server will close the connection (not returning a message): if we do get a message that is not that exact `authenticated` one, technically we are getting something that differs from the protocol. It's a good point though - I'll move it to the issue about polishing, as it would probably be easier for us if we got a message explicitly saying we are not authenticated (to differentiate from other cases).

Reply:
#101